### ISSUE TITLE
False positive in Style/TrivialAccessors with inline private

### ISSUE THREAD

**Issue by fujimura** [bbatsov/rubocop#5641]
## Expected behavior

Not an offense. In this case, `private` / `protected` clause needs symbol returned from `def` but `attr_reader` returns nil, which causes TypeError.

## Actual behavior

```ruby
# a.rb
class A
  private def foo
    @foo
  end
end
```

```
$ rubocop -a a.rb
Inspecting 1 file
C

Offenses:

a.rb:2:11: C: [Corrected] Style/TrivialAccessors: Use attr_reader to define trivial reader methods.
  private def foo
          ^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

```ruby
# Corrected a.rb
class A
  private attr_reader :foo
end
```

```
$ ruby a.rb
a.rb:2:in `private': nil is not a symbol nor a string (TypeError)
        from a.rb:2:in `<class:A>'
        from a.rb:1:in `<main>'
```

## Steps to reproduce the problem

Described in actual behavour.

## RuboCop version

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.2, running on ruby 2.4.2 x86_64-darwin16)
```

**Comment by Drenmi** [comment]
You can still define private readers using `attr_reader`:

```ruby
private

attr_reader :foo
```

although not using the new inline syntax.

**Comment by pocke** [comment]
I don't think it is a false positive for the same reason as the @Drenmi comment.
But the auto-correction behaviour is a bug. So we should disable the auto-correction on `private`.


BTW, maybe we can add an option to ignore this case, i.e. this cop ignores `private def foo ...` with this option. But I'm not sure the option is necessary as I think the `private; attr_reader :foo` is better style.

**Comment by fujimura** [comment]
> I don't think it is a false positive for the same reason as the @Drenmi comment. 
> But the auto-correction behaviour is a bug. So we should disable the auto-correction on private.

Agreed. Thanks for fixing this!

**Comment by Darhazer** [comment]
Fixed in 0.54. Can we close this?
